src/docs/arch-metadata.dox

```diff
@@ -1,19 +1,78 @@
+pairs in string form.  The key/value pairs in the configuration strings are dependent on the
+type of \c uri.  Thus, a metadata entry with a \c uri key beginning with \c "table:" will be
+a configuration string having entries such as \c key_format and \c value_format to
+describe the data encoding for the uri. A metadata key beginning with
 \c "file:" will have a different set of configuration entries associated
 with it.
 
-A caller of WiredTiger can use WT_SESSION::open_cursor with a \c uri equal to
-\c "meta:" to read the metadata.  Using this interface, metadata can only
-be queried, not changed.  Changes to the metadata are affected by API calls
-such as WT_SESSION::create, WT_SESSION::drop, WT_SESSION::rename.
+@section metadata_checkpoint Storing Checkpoint Metadata
+
+The latest checkpoint information for each btree in the system is stored in its metadata entry. The checkpoint metadata
+provides a known point in time from which WiredTiger can recover a btree's data after a restart.
+The checkpoint metadata for a given btree is stored within the configuration string under its
+associated \c uri key within the \c WiredTiger.wt table. Metadata configuration entries relevant to capturing
+a given btree's checkpoint state include:
+- \c checkpoint: The file's checkpoint entries, including their timestamp, transaction ID and write generation. Most importantly
+this information also tracks an encoded address cookie that describes the blocks that make up the checkpoint's data.
+- \c checkpoint_lsn: The LSN of the last checkpoint.
+- \c checkpoint_backup_info:  Incremental checkpoint backup information.
+
+@section metadata_turtle Turtle file
+
+The metadata file \c WiredTiger.wt is a regular WiredTiger table itself. As such, the metadata table has a btree that also needs to be
+checkpointed. Since a new metadata entry needs to be created for each btree at the time of checkpointing, the metadata table is the last
+file to be checkpointed in the process. Checkpointing the metadata table last is important
+for capturing a consistent metadata state of the btrees in the database at the time of the last checkpoint.
+
+Due to this ordering, to successfully restore the checkpoint of the \c WiredTiger.wt file, WiredTiger captures the most
+recent checkpointing information of the metadata table in a separate file. This being a special turtle file inside the db directory called
+\c "WiredTiger.turtle". The turtle file itself can be considered as being metadata for the WiredTiger metadata. Using the
+turtle file, WiredTiger can read the latest checkpoint information from the file when recovering the \c WiredTiger.wt table.
+Recovering the checkpoint of the \c WiredTiger.wt table in turn helps restore a consistent view of the checkpointing information
+for the other btrees in the system.
+
+@section metadata_reading Reading Metadata
+
+The contents of the metadata table is ASCII printable. The URIs and configuration values
+are all printable \c C strings.
```

Comment:
Remove `C`. They are just strings, not language-specific.

Reply:
Yep, good point. Removed in db14d9e1da5616d47c480feae5b4ae38a08cbb30